src/core/views.py

```diff
@@ -3559,14 +3714,143 @@ def view_proposal_review(request, proposal_id, assignment_id, access_key=None):
+        self.proposal_id = self.kwargs['proposal_id']
+        self.assignment_id = self.kwargs['assignment_id']
+        self.access_key = self.kwargs.get('access_key')
+
+        self.proposal = get_object_or_404(
+            submission_models.Proposal,
+            pk=self.proposal_id
+        )
+
+        if self.access_key:
+            self.proposal_reivew = get_object_or_404(
+                submission_models.ProposalReview,
+                pk=self.assignment_id,
+                access_key=self.access_key,
+            )
+        else:
+            self.proposal_reivew = get_object_or_404(
+                submission_models.ProposalReview,
+                pk=self.assignment_id,
+            )
+
+        return super(ProposalReviewCompletionEmail, self).dispatch(
+            request,
+            *args,
+            **kwargs
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super(ProposalReviewCompletionEmail, self).get_context_data(
+            **kwargs
+        )
+        context['heading'] = (
+            'Please ensure that you are happy with the below email to the '
+            'editor notifying them that you have completed your review'
+        )
+        return context
+
+    def get_form_kwargs(self):
+        """Renders the email body and subject for editing using the form."""
+        kwargs = super(ProposalReviewCompletionEmail, self).get_form_kwargs()
+
+        if (
+                self.proposal_reivew.requestor and
+                self.proposal_reivew.requestor.profile.full_name()
+        ):
+            recipient_greeting = u'Dear {assigning_editor_name}'.format(
+                assigning_editor_name=(
+                    self.proposal_reivew.requestor.profile.full_name()
+                )
+            )
+        else:
+            recipient_greeting = 'Dear sir or madam'
+
+        email_context = {
+            'greeting': recipient_greeting,
+            'proposal': self.proposal,
+            'sender': self.proposal_reivew.user,
+        }
+
+        email_body = email.get_email_content(
+            request=self.request,
+            setting_name='proposal_peer_review_completed',
+            context=email_context,
+        )
+        email_subject = (
+            u'Review completed - {title}'.format(
+                title=self.proposal.title,
+            )
+        )
+
+        kwargs['initial'] = {
+            'email_subject': email_subject,
+            'email_body': email_body,
+        }
+
+        return kwargs
+
+    def form_valid(self, form):
+        attachments = handle_multiple_email_files(
+            request_files=self.request.FILES.getlist('attachments'),
+            file_owner=self.proposal_reivew.user,
+        )
+
+        assigning_editor_email = ''
+        if self.proposal_reivew.requestor:
+            assigning_editor_email = (
+                self.proposal_reivew.requestor.email
+            )
+
+        book_editors_emails = [
```

Comment:
I wonder if the same resulting list could be achieved with a Django model query using filters, e.g. (beware: *untested code*)

```
self.proposal.book_editors.exclude(
    editor__email=assigning_editor_email,
    editor__username=settings.INTERNAL_USER
)
```